osquery/filesystem/filesystem_tests.cpp

```diff
@@ -69,12 +69,38 @@ TEST_F(FilesystemTests, test_wildcard_full_recursion) {
             files.end());
 }
 
+TEST_F(FilesystemTests, test_wildcard_end_last_component){
+  std::vector<std::string> files;
+  auto status = resolveFilePattern("/%bin/%sh", files);
+  EXPECT_TRUE(status.ok());
+  EXPECT_NE(std::find(files.begin(), files.end(), "/bin/bash"),
+            files.end());
+  EXPECT_NE(std::find(files.begin(), files.end(), "/bin/sh"),
+            files.end());
+}
+
+TEST_F(FilesystemTests, test_wildcard_three_kinds){
+  std::vector<std::string> files;
+  auto status = resolveFilePattern("/%sr/%/%%", files);
```

Comment:
i would assume something like this is pretty fast because you're specifying "sr"